### ISSUE TITLE
Questions or problems with drummer.land?

### ISSUE THREAD

**Issue by scripting** [scripting/drummerSupport#178]
I just posted the [announcement](http://scripting.com/2023/02/15.html#a155806) on Scripting News. 

If you have questions, or need help, post a message here. Include as much info as is needed for us to reproduce the problem. 

Dave

PS: This is where you go to [request your files be moved](http://scripting.com/code/readtweet/) from the old site to the new.

**Comment by fmcpherson** [comment]
Do you have a list of known problems? I don't want to double report. For example, is support for public URLs available or not ready yet?

**Comment by scripting** [comment]
Just report stuff that doesn’t work as it used to before the transition.

**Comment by dmatchett** [comment]
Thanks for all the work and sorry it was stressful. I understand that all users may not have accounts linked yet, but my experience was I used the "Sign in" menu item in drummer.land, received the email at my chosen address (the same address I use for Feedland), clicked the link in the email, and got a dialog in the browser: "The user "dmatchett@icloud.com" doesn't have an account on this server." I will try again later but just wanted to detail this -- cheers.

**Comment by scripting** [comment]
You have to sign up. You don’t have an account.

**Comment by dmatchett** [comment]
Very true -- I was distracted by the reference to Feedland, where I do have an account. Anyway: I see all my files so that's great. "Download my files" still works. A couple of things don't yet: the Safari bookmarklet invokes the Drummer notes tab, but doesn't create a new entry with the link. My number one hope is to have this working again: it's what made Drummer the fabulous tool it was. Hope you have time to tweak the bookmarklet! (I'm not a programmer, perhaps you can tell...) Thanks a million.

**Comment by scotthansonde** [comment]
@dmatchett The bookmarklet works just fine for me after updating the javascript URL to open https://drummer.land instead of http://drummer.scripting.com. Here's the long URL for the bookmarklet:

```
javascript: (function () %7B function encode (s) %7B s %3D s.toString ()%3B s %3D s.replace (new RegExp ("%25", "g"), " percent")%3B return (encodeURIComponent (s))%3B %7D var url %3D "https://drummer.land/%3F"%3B var what %3D "what%3Dbookmarklet"%3B var link %3D "%26link%3D" %2B encode (location.href)%3B var title %3D "%26title%3D" %2B encode (document.title)%3B var desc %3D "%26description%3D" %2B encode (document.getSelection ())%3B window.open (url %2B what %2B link %2B title %2B desc)%3B  %7D) ()%3B
```

**Comment by scripting** [comment]
@scotthansonde -- i did exactly that in an attempt to fix this problem but got a syntax error when i tried to run the bookmarklet. 

I'm going to take what you did and try putting it into the page, see what happens. And thanks for picking up the ball here. 

#### Update

The same thing happened when I pasted your bookmarklet into the web page. 

Also, it happened with the bookmarklet that was there before this whole experiment.

**Comment by akaKenSmith** [comment]
I went to use the fab Tools > Get my tweets feature of web Drummer just now, expecting to see today's tweets added to the open outline, as I've sometimes done in the past. "Get tweets you've posted today?" popped up, I clicked OK.

The next popup said, "There was an error getting your tweets." I clicked OK. No tweets were added to the outline. This happened when I tried again.

Over in the settings menu, the URL for my Twitter account is correct. Empty cache and hard reload didn't change what happens here on my laptop.

**Comment by scripting** [comment]
@akaKenSmith -- thanks for asking this question. all that stuff is due to break, depending entirely on what twitter does. i'm not going with them on this trip. as much as i like and use twitter myself, there's a cost to trying to remain compatible with a platform that doesn't want you there, and that's clearly where twitter is now. at some point the commands will be removed, probably.

**Comment by akaKenSmith** [comment]
> @akaKenSmith -- thanks for asking this question. all that stuff is due to break, depending entirely on what twitter does. i'm not going with them on this trip. as much as i like and use twitter myself, there's a cost to trying to remain compatible with a platform that doesn't want you there, and that's clearly where twitter is now. at some point the commands will be removed, probably.

Yes, that was pretty much my sad hunch. Get my tweets is/was a great tool, though. Back to the land of Cut and Paste we go.

**Comment by akaKenSmith** [comment]
I wanted to remind myself how Drummer/OldSchool presents an mp3 enclosure, so I created a [test posting](http://oldschool.scripting.com/ksmith3123@gmail.com/2023/04/09/235741.html). 

In the first line of the posting, I added an attribute, enclosure, and pasted in [an mp3 file](https://ksmith.files.wordpress.com/2023/03/washington-pledge-2023-32223-9.22-am.mp3) from my old WordPress account. Saved the attribute, rebuilt the blog, but no sign of the expected mp3 was visible there. 

I emptied the cache and glanced into the console during a hard reboot, nothing stood out to me there.

I reviewed the procedure steps and noticed that the mp3 file I was using has an https url. That's the only thing I've noticed that seems like it might be relevant.

**Comment by scripting** [comment]
@akaKenSmith -- you can help by, when reporting a problem, looking in the JavaScript console for error messages. We've been through this drill so many times.

**Comment by akaKenSmith** [comment]
> 

As I mentioned, i did look in the console and didn't see anything there that stood out as an error. I'll go back and take a screen capture.

OldSchool console screen shot of the 2 issues in blue and for Drummer, 1 issue in blue.

<img width="393" alt="Screenshot 2023-04-09 at 9 55 15 PM" src="https://user-images.githubusercontent.com/4759847/230810514-19f90967-b6ab-45c4-a2fa-41f0226026e4.png">
<img width="404" alt="Screenshot 2023-04-09 at 9 54 36 PM" src="https://user-images.githubusercontent.com/4759847/230810515-bb039f31-2bd8-42b5-a97d-019e00c3e043.png">

**Comment by scripting** [comment]
@akaKenSmith -- first, i appreciate that you reported an error, but the report didn't help me understand what the problem was. 

here's what I would consider a perfect report for the problem I'm guessing you encountered. 


####  Problem with web Drummer and enclosures
- Steps
-- Get the URL of an <a href="http://scripting.com/2023/04/07/howRssHappenedFromMyPov.m4a">MP3 file</a> on the clipboard.
-- In web Drummer, create a new headline, put the bar cursor on it, click the suitcase icon in the left margin.
-- Create a new attribute, enclosure, and save.
-- Wait a second or two.
-- Look in the JavaScript console.
-- You'll see an <a href="http://scripting.com/images/2023/04/10/mixedcontenterror.png">error message</a> saying there's mixed content. We're calling an HTTP routine when we are running in HTTPS.
- What this means
-- Automatic setting of enclosure length and type is not working in web Drummer.
-- Note it does work in Electric Drummer.

**Comment by scripting** [comment]
Update: I believe the problem is now fixed, that you can follow the steps in the previous message and the other enclosure data will correctly be filled in without error.

**Comment by akaKenSmith** [comment]
> Update: I believe the problem is now fixed, that you can follow the steps in the previous message and the other enclosure data will correctly be filled in without error.

@scripting, thanks for the enclosure data fix. Here as I use web Drummer:

1. Yesterday when I added an enclosure attribute with an mp3 file, as you note here, the other enclosure data was not produced and was not added as two new lines in the node's attribute listing. And as you say here, today those other two lines of attributes are now successfully added to the node's attribute listing after rebuilding the blog.

As far as I can tell, that portion of the process is fully functioning. **But I still can't find an audio enclosure in the new blog post, as described here:**

2. When I make a test posting with an mp3 enclosure [here](http://oldschool.scripting.com/ksmith3123@gmail.com/2023/04/10/222117.html) and view the item's OldSchool page source, I see that the mp3 url (starting in this case with https) has been successfully added as part of that posting in the html. Like this (in bold for clarity):

<img width="615" alt="Screenshot 2023-04-10 at 6 32 41 PM" src="https://user-images.githubusercontent.com/4759847/231014678-28966325-4e2c-4b98-aff4-8e5cfe74ad4b.png">

4. When I look at that test posting itself in OldSchool, I don't recognize any indication of an enclosure for an audio file. When I empty the cache and hard reload the page, I still don't recognize an enclosure for an audio file, as expected. 

<img width="516" alt="Screenshot 2023-04-10 at 7 07 03 PM" src="https://user-images.githubusercontent.com/4759847/231015833-b245f9ad-dcfb-4944-ab4d-3f6c1700549b.png">

**Drummer seems to have asked for the posting of an https enclosure and OldSchool seems to have declined?**

5. While most of what goes on in the developer's console is over my head, I don't see anything there today that I recognize as a new error message. Perhaps I'm not looking in the right place.

**Comment by scripting** [comment]
@akaKenSmith -- is this a bug report or is it working? if it's a bug report, what's the bug?

**Comment by akaKenSmith** [comment]
> @akaKenSmith -- is this a bug report or is it working? if it's a bug report, what's the bug?

Not sure why there's no step three in my edit, sorry. Step 2, Drummer inserts the https...mp3 file into the page source and asks OldSchool to post it as part of the blog page. Step 4, OldSchool posts the blog page but there is nothing in the blog post that I can recognize as an audio enclosure.

**Comment by scripting** [comment]
This is what a podcast looks like on my blog and yours. 

http://scripting.com/2023/04/07.html

So there’s no bug here. Let’s move on.
